core/src/main/java/org/apache/accumulo/core/file/FileOperations.java

```diff
@@ -48,38 +48,295 @@ public static FileOperations getInstance() {
     return new DispatchingFileFactory();
   }
 
+  //
+  // Abstract methods (to be implemented by subclasses)
+  //
+
+  protected abstract long getFileSize(GetFileSizeOperation options) throws IOException;
+
+  protected abstract FileSKVWriter openWriter(OpenWriterOperation options) throws IOException;
+
+  protected abstract FileSKVIterator openIndex(OpenIndexOperation options) throws IOException;
+
+  protected abstract FileSKVIterator openScanReader(OpenScanReaderOperation options) throws IOException;
+
+  protected abstract FileSKVIterator openReader(OpenReaderOperation options) throws IOException;
+
+  //
+  // File operations
+  //
+
   /**
-   * Open a reader that will not be seeked giving an initial seek location. This is useful for file operations that only need to scan data within a range and do
-   * not need to seek. Therefore file metadata such as indexes does not need to be kept in memory while the file is scanned. Also seek optimizations like bloom
-   * filters do not need to be loaded.
+   * Construct an operation object allowing one to query the size of a file. <br>
+   * Syntax:
    *
+   * <pre>
+   * long size = fileOperations.getFileSize().ofFile(filename, fileSystem, fsConfiguration).withTableConfiguration(tableConf).execute();
```

Comment:
OK, so I had a preformatted region, within a comment(!), and the auto formatter decided to reformat it?!?